### PR TITLE
Verify signatures of downloaded Kubernetes artifacts and bump versions

### PR DESCRIPTION
**Pull request by dnaeon**
**What this PR does / why we need it**:

This PR adds support for verifying the signatures of Kubernetes artifacts. 

It also:

- Bumps the `KUBERNETES_VERSION` to `v1.29.1`
- Bumps the `MINIMUM_KUBERNETES_VERSION` to `v1.27.0` (`v1.26.x` will be EOL'd at 28-02-2024, so `v1.27.x` will be the last supported release in about ~ 2 weeks)

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```improvement operator

```

### CODE REVIEW

**Review by vpnachev — COMMENTED** [review]
/reviewed ok-to-test

**Comment by dnaeon** [comment]
PR rebased.

**Review by vpnachev — CHANGES_REQUESTED** [review]

File: scripts/get-k8s-binary.sh [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+#  $ get-k8s-binary.sh kubelet v1.29.1 linux/amd64
+
+set -e
+
+BASE_URL=https://dl.k8s.io/release/
+
+function _main() {
+    if [ $# -ne 3 ]; then
+        echo "Usage: $0 <binary> <version> <os/arch>"
+        exit 64
```
[the remark this answers is not in the document]
Comment (dnaeon) [reply scripts/get-k8s-binary.sh]: The exit code is based on [sysexits.h(3)](https://manpages.debian.org/bookworm/manpages-dev/sysexits.h.3head.en.html), where `EX_USAGE (64)` specifies a command-line usage error.

**Pull request merged**